enhancements/api-review/gcp_user_defined_labels.md

```diff
@@ -0,0 +1,610 @@
+  openshift-* namespace. External operators are not in scope.
+  User-defined labels can be updated on the following GCP resources.
+  - Compute Engine Instance
+  - Compute Address
+  - Compute Image
+  - Compute Forwarding Rule
+  - Storage Bucket
+  - DNS Zone
+
+- GCP Labels can be used as queryable annotations for resources, but can't be used
+  to set conditions on policies.
+
+- User-defined tags cannot be updated on a GCP resource which is not managed by an
+  operator. In this proposal, the changes proposed and developed will be part of
+  openshift-* namespace. External operators are not in scope.
+  User-defined tags can be updated on the following GCP resources.
+  - Compute Engine Instances
+  - Cloud Storage Buckets
+
+- Administrator will have to manually perform below label pertaining actions
+    1. removing the undesired labels/tags from the required resources.
+    2. update label/tags values of the required resources.
+    3. update labels/tags of the resources which are not managed by an operator.
+    4. update labels/tags of the resources for which update logic is not supported
+       by an operator.
+
+## Design Details
+
+### Open Questions
+- GCP supports 50 tags per resource and since OpenShift do not add any tags of its own,
+  users are allowed to define upto 50 tags. But the terraform and GO APIs for binding the
+  tags to the resources doesn't support batch operation and requires each tag to be added
+  individually to each resource which will increase the cluster creation time. What would
+  be the optimal number of tags a user can define?
+
+  **[Solution]** : User can configure upto 50 tags. GCP has limit of 10 requests per second and
+  the operators will make use of rate limiting mechanism to limit the number requests
+  generated per second.
+  
+- GCP client libraries support two ways of providing tag details
+  1. NamespacedName: Tag value could be fed to APIs in the format
+     `<ORGANIZATION_ID>/<TAG_KEY_SHORT_NAME>/<TAG_VALUE_SHORT_NAME>` where
+      - ORGANIZATION_ID: Organization ID which is hosting the project under which OCP cluster
+     will be created and tag key, value resources exist.
+      - TAG_KEY_SHORT_NAME: Tag key short name is the display name provided during tag key
+     resource creation.
+      - TAG_VALUE_SHORT_NAME: Tag value short name is the display name provided during tag value
+     resource creation.
+  2. Name: Tag value could also be fed to the APIs in the format `tagValues/<TAG_VALUE_ID>` where
+      - TAG_VALUE_ID: Tag value ID is the generated numeric ID for the tag value resource using which
+     the tag key it is linked to can be identified.
+
+  GCP terraform API `google_tags_location_tag_binding` expects the tag value's `Name` whereas the GO API
+  `CreateTagBinding` accepts both tag value's `Name` and `NamespacedName`. The current approach for user
+  to input `ORGANIZATION_ID, TAG_KEY_SHORT_NAME, TAG_VALUE_SHORT_NAME` which has more readability, will require
```

Comment:
Are we expecting the installer to create the tags? Or the users will pass in references to already existing tags?

Reply:
Users have to pass the references to the tags which already exist at the Organization where the project in which OpenShift cluster will be created is present.